firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -150,10 +154,72 @@ def from_string(string):
             if invalid_character in string:
                 raise ValueError('Invalid characters in string.')
         string = string.split('.')
-        return FieldPath(*string)
+        return cls(*string)
+
+    @classmethod
+    def _parse_field_name(cls, api_repr):
```

Comment:
I think indexing a string `s[i]` is expensive in Python, because it creates a new string, right? So maybe use a regexp to find the parts?

By the way, I never needed this functionality in the Go client. When does it come up?

Reply:
Before you write any more code, let's figure out if you need `from_api_repr`. Where do you use it in the client?

Reply:
@jba I think having it would help me write code for tests that uses the actual api_repr, which is not ambiguous as in the case of using strings.  I know I could use the constructor instead, but it might not have been as clear.

Reply:
And I think it was also the `parse_field_name` function I could also use elsewhere.

Reply:
I thought about using regex, but I think it was easier for me to write it this way to get the behavior right.

Reply:
For tests, use the constructor to build the exact FieldPath you need. It doesn't make sense to write this complicated parsing code when you don't really need it.

Reply:
ok, closing this and moved the cleanups to #5121 and #5122.